src/quacc/recipes/vasp/core.py

```diff
@@ -15,6 +22,104 @@
     from quacc.schemas._aliases.vasp import DoubleRelaxSchema, VaspSchema
     from quacc.utils.files import Filenames, SourceDirectory
 
+logger = logging.getLogger(__name__)
+
+
+@job
+def nscf_job(
+    atoms: Atoms,
+    prev_dir: SourceDirectory,
+    bandgap: float | None = None,
+    nbands_factor: float | None = None,
+    preset: str | None = "BulkSet",
+    kpoints_mode: Literal["uniform", "line"] = "uniform",
+    calculate_optics: bool = False,
+    **calc_kwargs,
+) -> VaspSchema:
+    """
+    Carry out a non-self-consistent field (NSCF) calculation.
+
+    Parameters
+    ----------
+    atoms
+        Atoms object.
+    prev_dir
+        Directory (str | path) of the static_job
+    bandgap
+        The band gap of the material.
+    nbands_factor
+        A factor used to adjust NBANDS provided vasprun.xml(.gz) exists in
```

Comment:
A few suggestions:
1. Let's emphasize that it's a multiplicative factor
2. Let's set the default to 1.2
3. Let's make the type `float` instead of `float | None`. If someone doesn't want to increase it, they can set it to 1.0.